### PR TITLE
Fix #27 Add "path.properties" to list predicates

### LINKED ISSUE #27: Get all predicates of a resource
I need to get all predicates of a resource.

For instance, with the resource in the example (`https://ruben.verborgh.org/profile/#me`), I try:
```javascript
console.log(Object.keys(await person));
```
I expect:
```json5
['name', 'givenName', 'familyName', 'interest', 'friends', /*...*/ ]
```
but I get:
```json5
[ 'settings', 'subject', 'proxy', 'extendPath' ]
```

What is the good way to get all predicates of a resource with LDFlex?

### PR DESCRIPTION
**Pull request by jblemee**
Here is the pull request for #27 

SparqlHandler can now managed 3 new path parameters :
 - pathData.select : to set a specific select
 - pathData.distinct (boolean) : to add the DISTINCT keyword
 - pathData.finalClause : to add a final clause

These parameters are used by the new PropertiesHandler() list all predicates for a given path

### CODE REVIEW

File: src/PropertiesHandler.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Queries for all predicates of a path subject
+ *
+ */
+export default class PropertiesHandler {
+  handle(pathData) {
+    return pathData.extendPath({ finalClause: PropertiesHandler.finalClause, select: '?p', distinct: true, property: pathData.property });
+  }
+
+  static finalClause(queryVar) {
```
Comment (RubenVerborgh) [inline src/PropertiesHandler.js]: Member method instead of static?

File: src/SparqlHandler.js [hunk 2]
```diff
@@ -19,13 +19,21 @@ export default class SparqlHandler {
   }
 
   pathExpressionToQuery(pathData, path, pathExpression) {
-    if (pathExpression.length < 2)
+    if (pathExpression.length < 2 && !pathData.finalClause)
       throw new Error(`${pathData} should at least contain a subject and a predicate`);
 
+    let queryVar = '?s';
```
Comment (RubenVerborgh) [inline src/SparqlHandler.js]: Would make that `?subject`.

File: src/SparqlHandler.js [hunk 3]
```diff
@@ -19,13 +19,21 @@ export default class SparqlHandler {
   }
 
   pathExpressionToQuery(pathData, path, pathExpression) {
-    if (pathExpression.length < 2)
+    if (pathExpression.length < 2 && !pathData.finalClause)
```
Comment (RubenVerborgh) [inline src/SparqlHandler.js]: Nice integration with the existing code.

File: src/PropertiesHandler.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    return `\n  ${queryVar} ?p ?x.`;
```
Comment (RubenVerborgh) [inline src/PropertiesHandler.js]: Would go full here for debug purposes, so `?predicate ?object`.

File: src/PropertiesHandler.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Queries for all predicates of a path subject
+ *
+ */
+export default class PropertiesHandler {
```
Comment (RubenVerborgh) [inline src/PropertiesHandler.js]: Maybe we should call it `PredicatesHandler` internally for clarity (but just expose it through `.properties`—or perhaps additionally as `.predicates`).

**Comment by jblemee** [comment]
Thanks for the code review. I've done the changes

File: test/integration/sparql-test.js [hunk 6]
```diff
@@ -21,7 +21,8 @@ describe('a query path with a path expression handler', () => {
     pathExpression: new PathExpressionHandler(),
     add: new InsertFunctionHandler(),
     delete: new DeleteFunctionHandler(),
-    properties: new PropertiesHandler(),
+    properties: new PredicatesHandler(),
+    predicates: new PredicatesHandler(),
```
Comment (yukulele) [inline test/integration/sparql-test.js]: is it really necessary to keep both? wouldn't it be enough to keep only "predicate"?

Comment (RubenVerborgh) [reply test/integration/sparql-test.js]: The audience that LDflex targets might not be as familiar with the term "predicate".

**Comment by JordanShurmer** [comment]
Any update on this? I believe this was blocking #31, according to a comment on #31.

**Comment by RubenVerborgh** [comment]
Thanks a lot, apologies for the delay in merging.

**Pull request merged**